crates/rolldown/src/module_finalizers/scope_hoisting/mod.rs

```diff
@@ -27,6 +27,7 @@ pub struct ScopeHoistingFinalizer<'me, 'ast> {
   pub alloc: &'ast Allocator,
   pub snippet: AstSnippet<'ast>,
   pub comments: oxc::allocator::Vec<'ast, Comment>,
+  pub contains_reference_to_namespace_ref_of_this_module: bool,
```

Comment:
The biggest complexity comes from how we make `contains_reference_to_namespace_ref_of_this_module` right.

Reply:
Such as we need to consider if rewrited `member_expr` point to the `namespace_ref_of_this_module`.